examples/dbscan/reader.cpp

```diff
@@ -76,11 +78,184 @@ std::vector<ArborX::Point> loadArborXData(std::string const &filename,
+      lonlatno = 0;
+      pch = strtok_r(line, "\"[", &end_str);
+      while (pch != nullptr)
+      {
+        if (wordNo > 0)
+        {
+          char *pch2;
+          char *end_str2;
+
+          pch2 = strtok_r(pch, ",", &end_str2);
+
+          if (strcmp(pch2, "]") < 0 && lonlatno < 255)
+          {
+
+            thisWord = atof(pch2);
+
+            if (thisWord != 0.00000)
+            {
+              if (thisWord > -9 && thisWord < -7)
+              {
+                longitudes.push_back(thisWord);
+                // printf("lon %f",thisWord);
+                pch2 = strtok_r(nullptr, ",", &end_str2);
+                thisWord = atof(pch2);
+                if (thisWord < 42 && thisWord > 40)
+                {
+                  latitudes.push_back(thisWord);
+                  // printf(" lat %f\n",thisWord);
+
+                  lonlatno++;
+                }
+                else
+                {
+                  longitudes.pop_back();
+                }
+              }
+            }
+          }
+        }
+        pch = strtok_r(nullptr, "[", &end_str);
+        wordNo++;
+      }
+      // printf("num lonlat were %d x 2\n",lonlatno);
+    }
+    lineNo++;
+    if (lonlatno <= 0)
+    {
+      lineNo--;
+    }
+
+    // printf("Line %d\n",lineNo);
+  }
+  fclose(fp_data);
+
+  int num_points = longitudes.size();
+  assert(longitudes.size() == latitudes.size());
+  v.reserve(num_points);
+  for (unsigned int i = 0; i < num_points; ++i)
+    v.emplace_back(longitudes[i], latitudes[i], 0.f);
+
+  std::cout << "done\nRead in " << v.size() << " points" << std::endl;
+
+  return v;
+}
+
+std::vector<ArborX::Point> load3dSpatialNetworkData(std::string const &filename,
+                                                    bool binary)
+{
+  assert(!binary);
+  std::cout << "Trying to open " << filename
+            << " assuming 3dSpatialNetwork data.\n";
+  std::ifstream file(filename);
+  assert(file.good());
+
+  std::vector<ArborX::Point> v;
+
+  std::string thisWord;
+  while (file.good())
+  {
+    getline(file, thisWord, ',');
+    getline(file, thisWord, ',');
+    float longitude = stof(thisWord);
+    getline(file, thisWord, ',');
+    float latitude = stof(thisWord);
+    v.emplace_back(longitude, latitude, 0.f);
+  }
+  // really?????
+  // lon_ptr.pop_back();
+  // lat_ptr.pop_back();
+  std::cout << "done\nRead in " << v.size() << " points" << std::endl;
+
+  return v;
+}
+
 std::vector<ArborX::Point> loadData(std::string const &filename,
                                     std::string const &reader_type, bool binary)
 {
+  std::cout << "in reader" << std::endl;
   if (reader_type == "arborx")
     return loadArborXData(filename, binary);
+  else if (reader_type == "NGSIM")
+    return loadNGSIMData(filename, binary);
+  else if (reader_type == "TaxiPorto")
+    return loadTaxiPortoData(filename, binary);
+  else if (reader_type == "3dSpatialNetwork")
+    return load3dSpatialNetworkData(filename, binary);
```

Comment:
maybe we should change these for ease of using from the command line. How about "ngsim", "taxiporto" and "3droad"?